main.cpp

```diff
@@ -616,6 +659,43 @@ void part4()
     std::cout << "---------------------\n" << std::endl;
 }
 
+void part6()
+{
+    FloatType ft3(3.0f);
+    DoubleType dt3(4.0);
+    IntType it3(5);
+    
+    std::cout << "Calling FloatType::apply() using a lambda (adds 7.0f) and FloatType as return type:" << std::endl;
+    std::cout << "ft3 before: " << ft3 << std::endl;
+    ft3.apply( [&](float&)->FloatType&{ft3+=7.0f; return ft3;} );
```

Comment:
Don't write code like this.  this is impossible to read
curly braces go on their own line with no other code for a reason: It makes code readable.
- [ ] revise